pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,419 @@
+	pods, err := client.CoreV1().Pods(listPodFlags.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labelSelector.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	// define the filter callback function based on different flags
+	filterFunc := func(pod corev1.Pod) bool {
+		// filter by namespace if specified
+		if listPodFlags.Namespace != "" && listPodFlags.Namespace != pod.Namespace {
+			return false
+		}
+		// filter by jobName if specified
+		if listPodFlags.JobName != "" && listPodFlags.JobName != pod.Labels[v1alpha1.JobNameKey] {
+			return false
+		}
+		// filter by QueueName if specified
+		if listPodFlags.QueueName != "" && listPodFlags.QueueName != pod.Labels[v1alpha1.QueueNameKey] {
+			return false
+		}
+		return true
+	}
+	filteredPods := filterPods(pods, filterFunc)
+
+	if len(filteredPods.Items) == 0 {
+		fmt.Printf("No resources found\n")
+		return nil
+	}
+	PrintPods(filteredPods, os.Stdout)
+
+	return nil
+}
+
+func PrintPods(pods *corev1.PodList, writer io.Writer) {
+	maxNameLen := 0
+	maxReadyLen := 0
+	maxStatusLen := 0
+	maxRestartLen := 0
+	maxAgeLen := 0
+
+	var infoList []PodInfo
+	for _, pod := range pods.Items {
+		info := printPod(&pod)
+		infoList = append(infoList, info)
+		// update max length for each column
+		if len(info.Name) > maxNameLen {
+			maxNameLen = len(info.Name)
+		}
+		if len(info.ReadyContainers) > maxReadyLen {
+			maxReadyLen = len(info.ReadyContainers)
+		}
+		if len(info.Status) > maxStatusLen {
+			maxStatusLen = len(info.Status)
+		}
+		if len(info.Restarts) > maxRestartLen {
+			maxRestartLen = len(info.Restarts)
+		}
+		if len(info.CreationTimestamp) > maxAgeLen {
+			maxAgeLen = len(info.CreationTimestamp)
+		}
+	}
+	columnSpacing := 8
+	maxNameLen += columnSpacing
+	maxReadyLen += columnSpacing
+	maxStatusLen += columnSpacing
+	maxRestartLen += columnSpacing
+	maxAgeLen += columnSpacing
+	formatStr := fmt.Sprintf("%%-%ds%%-%ds%%-%ds%%-%ds%%-%ds\n", maxNameLen, maxReadyLen, maxStatusLen, maxRestartLen, maxAgeLen)
+	_, err := fmt.Fprintf(writer, formatStr, Name, Ready, Status, Restart, Age)
+	if err != nil {
+		fmt.Printf("Failed to print Pod information: %s.\n", err)
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// filterPods filters pods based on the provided filter callback function.
+func filterPods(pods *corev1.PodList, filterFunc func(job corev1.Pod) bool) *corev1.PodList {
+	filteredPods := &corev1.PodList{}
+	for _, pod := range pods.Items {
+		if filterFunc(pod) {
+			filteredPods.Items = append(filteredPods.Items, pod)
+		}
+	}
+	return filteredPods
+}
+
+// createLabelSelector creates a label selector based on the provided job name or queue name.
+func createLabelSelector(jobName, queueName string) (labels.Selector, error) {
```

Comment:
here @Monokaix Here is the filter logic of the selector

Reply:
Ok, got that, but label selector has already specified the queue and job name, is it necessary to filter pods another time by queue and job name?

Reply:
Seems that you have already considered init containers: )

Reply:
done